### DISCUSSION TITLE
Next.js 15 support

### DISCUSSION THREAD

**Discussion by martpie** [avitorio/outstatic#254] [Ideas]
https://nextjs.org/blog/next-15

It seems that now a bunch of things require `await` like URL params. Things sometimes still work in dev mode, but fail when trying to build a Next.js app.

Can we imagine supporting Next.js 15 soon? :)

**Comment by avitorio** [comment]
Hey @martpie 
Thank you for the heads up. I am still catching up with Next.js 15 but will make the necessary changes so that Outstatic 2.0 supports it!

**Comment by avitorio** [comment]
Hey @martpie - Just wanted to let you know that outstatic v1.4.12 offers Next.js 15 support!

**Reply by martpie** [reply]
I am getting the following error when trying to build a website `npm run build`

```tsx
src/app/api/outstatic/[[...ost]]/route.ts
Type error: Route "src/app/api/outstatic/[[...ost]]/route.ts" has an invalid "GET" export:
  Type "GetParams" is not a valid type for the function's second argument.

   Linting and checking validity of types  ..error: script "build" exited with code 1
```

did you face something similar?

**Reply by martpie** [reply]
I guess it's probably because the API must return async params, currently, they are `(property) POST: (req: Request, { params }: PostParams) => Promise<Response>`

**Reply by avitorio** [reply]
Hey @martpie , sorry about that. v1.4.13 fixes it. 
Thank you for letting me know!

**Reply by martpie** [reply]
Working and deployed, thank you very much!
